ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/ServerInfoResource.java

```diff
@@ -25,15 +30,41 @@
 @Path("/info")
 @Produces({Versions.KSQL_V1_JSON, MediaType.APPLICATION_JSON})
 public class ServerInfoResource {
+  private static final long DESCRIBE_CLUSTER_TIMEOUT_SECONDS = 30;
 
-  private final io.confluent.ksql.rest.entity.ServerInfo serverInfo;
+  private final ServiceContext serviceContext;
+  private final KsqlConfig ksqlConfig;
+  private volatile io.confluent.ksql.rest.entity.ServerInfo serverInfo = null;
 
-  public ServerInfoResource(final io.confluent.ksql.rest.entity.ServerInfo serverInfo) {
-    this.serverInfo = serverInfo;
+  public ServerInfoResource(final ServiceContext serviceContext, final KsqlConfig ksqlConfig) {
+    this.serviceContext = serviceContext;
+    this.ksqlConfig = ksqlConfig;
+  }
+
+  private ServerInfo getServerInfo() {
+    if (serverInfo == null) {
+      serverInfo = new ServerInfo(
+          Version.getVersion(),
+          getKafkaClusterId(serviceContext),
+          ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG)
+      );
+    }
+    return serverInfo;
+  }
+
+  private static String getKafkaClusterId(final ServiceContext serviceContext) {
```

Comment:
Could this be part of `KafkaClusterUtil` with an `AdminClient` as a parameter?

Reply:
Actually I'm not sure there's much to be gained from making this common. It's a pretty straightforward operation, and most of the interesting stuff is specific to how the operation is done here (converting exceptions to RuntimeException (which should turn into a 5xx response, the timeout).

Reply:
True, the amount of code to use either the `KafkaClusterUtil.getKafkaClusterId(adminClient)` or the `adminClient.describeCluster.clusterId().get()` is the same. Both throw exceptions which I should catch. 

Sounds fair, but it also helps to keep the Kafka cluster utility methods organized in one single class. It's optional, though. if you want to keep it or move it. I'm good on any choice.